server/src/test/java/io/unitycatalog/server/base/externallocation/BaseExternalLocationCRUDTest.java

```diff
@@ -0,0 +1,100 @@
+package io.unitycatalog.server.base.externallocation;
+
+import static io.unitycatalog.server.utils.TestUtils.COMMENT;
+import static io.unitycatalog.server.utils.TestUtils.COMMENT2;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.unitycatalog.client.ApiException;
+import io.unitycatalog.client.model.*;
+import io.unitycatalog.server.base.BaseCRUDTest;
+import io.unitycatalog.server.base.ServerConfig;
+import io.unitycatalog.server.base.storagecredential.StorageCredentialOperations;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public abstract class BaseExternalLocationCRUDTest extends BaseCRUDTest {
+  private static final String EXTERNAL_LOCATION_NAME = "uc_testexternallocation";
+  private static final String NEW_EXTERNAL_LOCATION_NAME = EXTERNAL_LOCATION_NAME + "_new";
+  private static final String URL = "s3://unitycatalog-test";
+  private static final String NEW_URL = "s3://unitycatalog-test-new";
+  private static final String CREDENTIAL_NAME = "uc_testcredential";
+  private static final String DUMMY_ROLE_ARN = "arn:aws:iam::123456789012:role/role-name";
+  protected ExternalLocationOperations externalLocationOperations;
+
+  protected abstract ExternalLocationOperations createExternalLocationOperations(
+      ServerConfig config);
+
+  protected StorageCredentialOperations storageCredentialOperations;
+
+  protected abstract StorageCredentialOperations createStorageCredentialOperations(
+      ServerConfig config);
+
+  @BeforeEach
+  @Override
+  public void setUp() {
+    super.setUp();
+    externalLocationOperations = createExternalLocationOperations(serverConfig);
+    storageCredentialOperations = createStorageCredentialOperations(serverConfig);
+  }
+
+  @Test
+  public void testExternalLocationCRUD() throws ApiException {
+    // Create an external location
+    CreateExternalLocation createExternalLocation =
+        new CreateExternalLocation()
+            .name(EXTERNAL_LOCATION_NAME)
+            .comment(COMMENT)
+            .url(URL)
+            .credentialName(CREDENTIAL_NAME);
+    // Fails as the credential does not exist
+    assertThatThrownBy(
+            () -> externalLocationOperations.createExternalLocation(createExternalLocation))
+        .isInstanceOf(ApiException.class);
```

Comment:
can we not test out the exception message. so that we are sure that we are throwign the right error that its indeed because creds dont exist.